src/components/Chip.tsx

```diff
@@ -18,27 +18,47 @@ import * as React from 'react';
 import { createStyleSelector } from '../helpers/styleCreators';
 
 interface Props {
-	text: string | number;
+	text?: string | number;
+	className?: string;
+	additionalClassName?: string;
 	title?: string;
 	isSelected?: boolean;
 	isLoading?: boolean;
 	onClick?: (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
+	onMouseEnter?: (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
+	onMouseLeave?: (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
+	children?: React.ReactNode;
 }
 
 export function Chip(props: Props) {
-	const { text, isSelected, onClick, title, isLoading } = props;
+	const {
+		text,
+		className,
+		additionalClassName,
+		isSelected,
+		onClick,
+		title,
+		isLoading,
+		onMouseEnter,
+		onMouseLeave,
+		children,
+	} = props;
 	const rootClass = createStyleSelector(
 		'chip',
 		isSelected ? 'selected' : null,
 		onClick ? 'clickable' : null,
 		isLoading ? 'loading' : null,
+		additionalClassName || null,
 	);
 
 	return (
-		<div className={rootClass} title={title} onClick={e => onClick && onClick(e)}>
-			<div className='chip__title'>
-				<p>{text}</p>
-			</div>
+		<div
+			className={className || rootClass}
```

Comment:
if someone provies className he would have reimplemented all the styles, it's better to concat additional class to rootClass than replacing it